docs/olmv1_mvp.md

```diff
@@ -0,0 +1,136 @@
+
+**B5 - Force overrides (P1):** While OLM has a lot of opinions about safe operations with cluster extensions they do not apply all the time since OLM cannot possibly foresee how extensions behave at runtime. It needs to yield to the user in cases where they have more certainty about what's going to happen based on their background knowledge of the cluster or the operator. It should offer ways to force-override decisions that OLM made that block the user from proceeding in a certain direction, especially in the areas of extension installation, removal and updates.
+
+**B6 - Human-readable status extensions information (P2):** Whenever OLM is in the process of or having reached or failed to reach a desired state it needs to update the user about what is happening / what has happened without assuming knowledge about OLM internal or implementation details.
+
+**B7 - Scalability & Resource consumption (P1):** OLM is used on clusters with hundreds to thousands of namespaces and tenants. Its API controls, specifically for F2 and F12 need to be built in such a way that resource consumption scales linearly with usage and cluster size and the overall resource usage envelope stays within manageable bounds that does not put the cluster stability, especially that of the API server at risk. System memory especially is a scarce resource.
+
+# Compatibility Requirements:
+
+**C1 - Compatibility with existing extensions  (P1):** OLM should be able to manage extensions packaged with the current bundle format in the way described by the functional and behavior requirements when the bundle supports AllNamespace installation mode.
+
+**C2 - Compatibility with existing catalogs (P1):** OLM should be able to retrieve and update extensions that adhere to C1 from the currently supported catalog formats (File-based catalogs).
+
+**C3 - Incompatibility with existing extensions  (P1):** OLM 1.0 does not support managing bundles or extension versions that do not support AllNamespace installation mode with the new set of APIs or flows
+
+## Assumptions
+- No additional tenancy model will be introduced at the control plane / API layer of Kubernetes upstream
+
+- kcp doesn’t fundamentally change OLMs role and responsibilities around managing extensions (at least initially)
+
+- OLM will move to a descoped, cluster-wide singleton model for cluster extensions, extension management isn’t namespaced
+
+## Constraints
+- Only operator bundles with “AllNamespace” mode installation support can be lifecycled with the new APIs / flows in OLM 1.0
+
+## Dependencies
+- RBAC templating (OLM-2158) and discrete visibility control (OLM-2366) will land prior to OLM 1.0 and unblock most operators that do not support AllNamespace installation mode today
+
+- Bandwidth in the Console engineering team to surface the new flows / forms / visualizations associated with any functional requirements
+
```

Comment:
Are these relevant to include in the doc? These seem to have some references to Jira tickets and internal Red Hat teams. Also I'm not sure the RBAC templating is still a dependency (could be wrong here).

Reply:
Right, this PR was intended to be a direct porting; hence, I didn't look into the details, so thanks for catching this.

The first bullet here is meant to call out: for OLM 1.0 to succeed, it is crucial to address "F13 - Extension permissions management (P1)" and "F12 - Installed extension discovery (P1)" first so cluster admins can grant Operators the minimum necessary permissions to fulfill their intended function rather than always granting permissions at the cluster level.  I'll remove those old Jira ticket IDs and reword them, and I'll remove the console-related work.